packages/tools/src/tools/base/ContourSegmentationBaseTool.ts

```diff
@@ -0,0 +1,355 @@
+    };
+
+    for (let i = 0; i < annotations.length; i++) {
+      const annotation = annotations[i] as Annotation;
+
+      styleSpecifier.annotationUID = annotation.annotationUID;
+
+      const annotationStyle = this.getAnnotationStyle({
+        annotation,
+        styleSpecifier,
+      });
+
+      const annotationRendered = this.renderAnnotationInstance({
+        enabledElement,
+        targetId,
+        annotation,
+        annotationStyle,
+        svgDrawingHelper,
+      });
+
+      renderStatus ||= annotationRendered;
+    }
+
+    return renderStatus;
+  }
+
+  /**
+   * Check if an annotation is a contour segmentation annotation.
+   *
+   * This is helpful on classes that can work in a hybrid way as a normal
+   * annotation tool or as a segmentation tool such as spline ROI tool. In that
+   * the segmentation must not be registered/unregistered as a contour
+   * segmentation annotation.
+   *
+   * @param annotation - Annotation
+   * @returns True if the annotation is a contour segmentation annotation or false otherwise
+   */
+  protected isSegmentationAnnotation(
+    annotation: Annotation
+  ): annotation is SegmentationAnnotation {
+    return (
+      (annotation as SegmentationAnnotation).segmentationData !== undefined
+    );
+  }
+
+  /**
+   * Return the segmentation data that needs to be merged with annotation data
+   * in any child class to compose the annotation data.
+   * @returns Annotation segmentation data
+   */
+  protected getSegmentationAnnotationData(): SegmentationAnnotationData {
+    const { toolGroupId } = this;
+    const activeSegmentationRepresentation =
+      activeSegmentation.getActiveSegmentationRepresentation(toolGroupId);
+
+    if (!activeSegmentationRepresentation) {
+      throw new Error(
+        'No active segmentation detected, create one before using scissors tool'
+      );
+    }
+
+    const { type: segmentationType } = activeSegmentationRepresentation;
+
+    if (segmentationType !== SegmentationRepresentations.Contour) {
+      throw new Error(`A contour segmentation must be active`);
+    }
+
+    const { segmentationId, segmentationRepresentationUID } =
+      activeSegmentationRepresentation;
+    const segmentIndex =
+      segmentIndexController.getActiveSegmentIndex(segmentationId);
+
+    return {
+      segmentationData: {
+        segmentationId,
+        segmentIndex,
+        segmentationRepresentationUID,
+      },
+    };
+  }
+
+  protected addAnnotation(annotation: Annotation, element: HTMLDivElement) {
+    const isSegmentationAnnotation = this.isSegmentationAnnotation(annotation);
+
+    // annotationUID is available only after calling addAnnotation
+    addAnnotation(annotation, element);
+
+    if (isSegmentationAnnotation) {
+      this._registerContourSegmentationAnnotation(annotation);
+    }
+  }
+
+  /**
+   * Unregister the segmentation when the annotation is canceled
+   * @param annotation - Contour segmentation annotation
```

Comment:
Suggest documentation by naming rather than a @param docs - so, use contourSegmentationAnnotation for the name, or perhaps (shorter) contourSegmentation
I do appreciate the docs on this one because it does something slightly different than most other cancels